### ISSUE TITLE
[Bug Report]: ReferenceError: __filename is not defined

### ISSUE THREAD

**Issue by felixmosh** [web-infra-dev/rspack#2275]
### System Info

System:
    OS: macOS 13.2.1
    CPU: (12) x64 Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz
    Memory: 345.54 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.16.0 - ~/Library/Caches/fnm_multishells/8168_1678870975199/bin/node
    Yarn: 1.22.19 - ~/.yarn/bin/yarn
    npm: 8.11.0 - ~/Library/Caches/fnm_multishells/8168_1678870975199/bin/npm
    Watchman: 2023.02.06.00 - /usr/local/bin/watchman
  Browsers:
    Chrome: 111.0.5563.64
    Firefox: 111.0
    Safari: 16.3

### Details

When bundling a code such as
```js
const highlightWorker = new Worker(
  /* webpackChunkName: "highlight-worker" */ new URL('./worker.ts', import.meta.url)
);
```

it generates a code that using __filename even though the `target` is web.
```js
a = new Worker(new URL("./worker.ts",n("url").pathToFileURL(__filename).toString())))
```

### Reproduce Steps

create a bundle with an entry of 
```js
const highlightWorker = new Worker(
  /* webpackChunkName: "highlight-worker" */ new URL('./worker.ts', import.meta.url)
);
```

**Comment by underfin** [comment]
[Here](https://github.com/web-infra-dev/rspack/blob/main/crates/rspack_plugin_javascript/src/visitors/dependency/scanner.rs#L362) is `__dirname` implement.

**Comment by sladyn98** [comment]
In the __dirname implementation we have a codeblock that could be generating the filename, is there a way to determine what is causing the error. As a first time contributor looking at the code the .relative function could be returning an invalid path as well. I am not quite sure though. 
```
"true" => Some(
                self
                  .resource_data
                  .resource_path
                  .parent()
                  .expect("TODO:")
                  .relative(self.compiler_options.context.as_ref())
                  .to_string_lossy()
                  .to_string(),
              ),
              ```

**Comment by underfin** [comment]
closed at #2342

**Comment by underfin** [comment]
@sladyn98 I'm also not sure about this, can you open a new issue to track this? Thanks a lot!
